### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by daandemeyer**
Split out of #32445

### CODE REVIEW

File: test/README.testsuite [hunk 1]
```diff
@@ -31,26 +31,26 @@ $ sudo make -C test/TEST-01-BASIC clean setup run
...
 
-$ meson test -C build/ -v TEST-01-BASIC
+$ SYSTEMD_INTEGRATION_TESTS=1 meson test -C build/ -v TEST-01-BASIC
 
 Due to limitations in meson, the integration tests do not yet depend on the mkosi target, which means the
 mkosi target has to be manually rebuilt before running the integration tests. To rebuild the image and rerun
 a test, the following command can be used:
 
-$ meson compile -C build mkosi && meson test -C build -v TEST-01-BASIC
+$ SYSTEMD_INTEGRATION_TESTS=1 meson compile -C build mkosi && meson test -C build -v TEST-01-BASIC
```
Comment (richardmaw-codethink) [inline test/README.testsuite]: environment setting binds only to the command not the whole group
```suggestion
$ meson compile -C build mkosi && SYSTEMD_INTEGRATION_TESTS=1 meson test -C build -v TEST-01-BASIC
```

File: src/udev/udevd.c [hunk 2]
```diff
@@ -289,18 +289,14 @@ int run_udevd(int argc, char *argv[]) {
...
+        log_setup();
 
         manager = manager_new();
         if (!manager)
                 return log_oom();
 
         manager_parse_udev_config(manager);
 
-        log_parse_environment();
-        log_open(); /* Done again to update after reading configuration. */
```
Comment (YHNdnzj) [inline src/udev/udevd.c:302]: Why remove this? The log target might have changed after parsing udev.conf

**Review by YHNdnzj — CHANGES_REQUESTED** [review]

File: src/udev/udevd.c (hunk 2 above)
Comment (daandemeyer) [reply src/udev/udevd.c:302]: @YHNdnzj We don't seem to read a log target from udev.conf? See `manager_parse_udev_config()`

**Comment by poettering** [comment]
lgtm

File: src/udev/udevd.c (hunk 2 above)
Comment (YHNdnzj) [reply src/udev/udevd.c:302]: Hmm, right. Sorry for the noise.

**Review by bluca — APPROVED** [review]

**Pull request merged**